magefiles/magefile.go

```diff
@@ -496,6 +496,13 @@ func (ci CI) setRequiredEnvVars() error {
 			if !strings.Contains(jobName, "rehearse") {
 				os.Setenv(fmt.Sprintf("%s_CATALOG_URL", envVarPrefix), fmt.Sprintf("https://github.com/%s/%s", pr.RemoteName, pr.RepoName))
 				os.Setenv(fmt.Sprintf("%s_CATALOG_REVISION", envVarPrefix), pr.CommitSHA)
+				if ci.isPRPairingRequired("release-service") {
+					sp := strings.Split(os.Getenv("COMPONENT_IMAGE"), "@")
```

Comment:
In the CI config of release-service-catalog repo,  there is no "COMPONENT_IMAGE" setting.  In paired PRs of "release-service" and "release-service-catalog",  I guess the value of "COMPONENT_IMAGE" from release-service repo is expected here, right?  I am not sure if the value be got as expected? @psturc, can you please help to confirm it?

Reply:
that's correct, the `COMPONENT_IMAGE` env var [does not exist in release-service-catalog job](https://github.com/openshift/release/blob/master/ci-operator/config/redhat-appstudio/release-service-catalog/redhat-appstudio-release-service-catalog-development.yaml), so IMO this won't work

if we want to pair release-service-catalog PR with release-service PR and use the container image built from the release-service PR, we could do that by using [the image built by tekton-ci](https://quay.io/repository/redhat-user-workloads/rhtap-release-2-tenant/release-service/release-service?tab=tags&tag=latest), i.e. we need a function here that extracts the head commit SHA from the related release-service PR and uses it as a tag:
```go
os.Setenv(fmt.Sprintf("%s_IMAGE_REPO", envVarPrefix), "quay.io/redhat-user-workloads/rhtap-release-2-tenant/release-service/release-service")
os.Setenv(fmt.Sprintf("%s_IMAGE_TAG", envVarPrefix), fmt.Sprintf("on-pr-%s", commitSHA))
```
drawback of this approach is that the openshift-ci would rely on tekton-ci to successfully build and push the image

other option I can think of (or a fallback option, in case the image built by tekton wouldn't exist) could be to build the release-service image directly on the testing cluster (using konflux app/component model in BeforeAll node of the release-service-catalog test)

Reply:
Would it make sense to add that env var to the release-service-catalog job? Apologies if that doesn't make sense to do.. I am not super well versed in how this stuff is configured. The goal is for most of the time, have the release-service-catalog CI run the release-pipeline suite with whatever release-service image is merged in infra, but to allow pairing in case a release-service-catalog PR requires an unmerged change in release-service

Reply:
Currently the pairing between e2e-tests to the other repos can be supported from [link](https://github.com/redhat-appstudio/e2e-tests/blob/main/docs/Installation.md#rhtap-in-openshift-ci-and-branch-pairing).  We need to confirm if the PR in release-service repo can pair the the PR in release-service-catalog PR.

Reply:
Yeah, I was the one who wrote a bit of that linked text. A catalog PR won't pair with a service PR right now I don't think, at least judging by the magefile. That is what I am trying to enable

Reply:
@psturc coming back to this, your suggestion sounds like a good one to use the container image built from the release service PR. I updated the PR for this. Is this all we would need? When I document this, I could just write that the pairing depends on the `ci/prow/images` check on the release-service pr passing first